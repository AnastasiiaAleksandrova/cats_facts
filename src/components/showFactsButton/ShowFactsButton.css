.more-facts {
    margin-top: 2em;
}

.more-facts-text {
    font-weight: 300;
    font-size: .8em;
    margin: .5em;
}