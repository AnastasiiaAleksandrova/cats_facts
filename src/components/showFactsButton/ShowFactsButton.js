import React from "react";
import { Button } from "@material-ui/core";
import "./ShowFactsButton.css";


function ShowFactsButton(props) {
    return (
        <div className="more-facts">
            <div className="more-facts-text">Have not decided yet...</div>
            <div>
                <Button
                    variant="contained"
                    color="primary"
                    onClick={props.onClick}>
                    Show me other facts
                </Button>
            </div>
        </div>
    );
}

export default ShowFactsButton;
