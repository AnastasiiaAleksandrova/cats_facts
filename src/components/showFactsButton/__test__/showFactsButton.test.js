import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import renderer from 'react-test-renderer';
import ShowFactsButton from "../ShowFactsButton";


let container = null;

beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

const dummyOnClick = jest.fn();
dummyOnClick.mockReturnValue(true);


it("renders without crashing", () => {
    render(<ShowFactsButton onClick={dummyOnClick} />, container);
});

it("calls a callback from props", () => {
    render(<ShowFactsButton onClick={dummyOnClick} />, container);
    const button = document.getElementById("show-more-facts");
    button.dispatchEvent(new MouseEvent("click", { bubbles: true }));
    expect(dummyOnClick).toHaveBeenCalledTimes(1);
});

it("matches snapshot", () => {
    const component = renderer.create(<ShowFactsButton />).toJSON();
    expect(component).toMatchSnapshot();
})



