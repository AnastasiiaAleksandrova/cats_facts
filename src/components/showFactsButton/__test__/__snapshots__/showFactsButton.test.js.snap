// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="more-facts"
>
  <div
    className="more-facts-text"
  >
    Have not decided yet...
  </div>
  <div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      id="show-more-facts"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Show me other facts
      </span>
    </button>
  </div>
</div>
`;
